backend/database.js

```diff
@@ -184,39 +205,42 @@ function throwIfFileOutsideDataDir (filename: string): string {
   return filepath
 }
 
-if (production || process.env.GI_PERSIST) {
-  // https://github.com/isaacs/node-lru-cache#usage
-  const cache = new LRU({
-    max: Number(process.env.GI_LRU_NUM_ITEMS) || 10000
-  })
-
-  sbp('sbp/selectors/overwrite', {
-    // we cannot simply map this to readFile, because 'chelonia/db/getEntry'
-    // calls this and expects a string, not a Buffer
-    // 'chelonia/db/get': sbp('sbp/selectors/fn', 'backend/db/readFile'),
-    'chelonia/db/get': async function (filename: string) {
-      const lookupValue = cache.get(filename)
-      if (lookupValue !== undefined) {
-        return lookupValue
-      }
-      const bufferOrError = await sbp('backend/db/readFile', filename)
-      if (Boom.isBoom(bufferOrError)) {
-        return null
-      }
-      const value = bufferOrError.toString('utf8')
-      cache.set(filename, value)
-      return value
-    },
-    'chelonia/db/set': async function (filename: string, data: any): Promise<*> {
-      // eslint-disable-next-line no-useless-catch
-      try {
-        const result = await sbp('backend/db/writeFile', filename, data)
-        cache.set(filename, data)
-        return result
-      } catch (err) {
-        throw err
+export default async () => {
+  // If persistence must be enabled:
+  // - load and initialize the selected storage backend
+  // - register `readString` and `writeString` selectors
+  // - overwrite 'chelonia/db/get' and '-set' to use an LRU cache
+  if (persistence) {
+    const { initStorage, readString, writeString } = await import(`./${persistence}-backend.js`)
```

Comment:
One request: could you please rename the files to have this format? `database-{persistence}.js`?

This will keep them sorted next to each other nicely in file viewers and code editors.